src/lib/database/store.ts

```
@@ -875,5 +927,35 @@ const initializer = immer<Database>((set, get) => ({
     )
   },
 
+  getEndpoints(params) {
+    const client_session = get().client_sessions.find(
+      (cs) => cs.client_session_id === params.client_session_id,
+    )
+
+    if (client_session?.user_identity_id === undefined) {
+      throw new Error(
+        "Could not find client session associated with a user identity!",
+      )
+    }
```

should we throw an unauthorized error here?